service/src/test/java/bio/terra/workspace/service/workspace/AwsCloudContextConnectedTest.java

```diff
@@ -0,0 +1,179 @@
+
+    // Discover the environment.  This should be done at the scope of a single API operation;
+    // caching will happen under the hood to prevent too many S3 calls to the discovery bucket.
+    Environment environment = environmentDiscovery.discoverEnvironment();
+    assertNotNull(environment.getMetadata(), "environment.metadata null");
+    assertNotNull(
+        environment.getWorkspaceManagerRoleArn(), "environment.workspaceManagerRoleArn null");
+    assertNotNull(environment.getUserRoleArn(), "environment.userRoleArn null");
+    assertNotNull(environment.getNotebookRoleArn(), "environment.notebookRoleArn null");
+    assertFalse(environment.getSupportedRegions().isEmpty(), "environment.supportedRegions empty");
+
+    // This lifetime should track that of the discovered environment where possible (stale
+    // credentials will get refreshed under the hood).
+    AwsCredentialsProvider awsCredentialsProvider =
+        AwsUtils.createWsmCredentialProvider(awsConfiguration.getAuthentication(), environment);
+
+    // Log details about the discovered environment.
+    logMetadata(environment.getMetadata(), "AWS Environment Infrastructure Details:");
+
+    // Iterate over supported regions in Environment.
+    for (Region region : environment.getSupportedRegions()) {
+      Optional<LandingZone> landingZoneOptional = environment.getLandingZone(region);
+      assertTrue(landingZoneOptional.isPresent());
+
+      LandingZone landingZone = landingZoneOptional.get();
+      assertNotNull(landingZone.getMetadata(), region + ": landingZone.metadata null");
+      assertNotNull(landingZone.getStorageBucket(), region + ": landingZone.storageBucket null");
+      assertNotNull(landingZone.getKmsKey(), region + ": landingZone.kmsKey null");
+      assertFalse(
+          landingZone.getNotebookLifecycleConfigurations().isEmpty(),
+          region + ": landingZone.notebookLifecycleConfigurations empty");
+
+      // Log details about the landing zone.
+      logMetadata(landingZone.getMetadata(), "Landing zone region: " + region);
+
+      // Perform a HeadBucket command to check storage bucket existence; this requires ListBucket
+      // permissions on the bucket, which the WSM role has.  This is a basic "Hello, world!" test
+      // that we've discovered the environment and assumed the TerraWorkspaceManager role.
+
+      String bucketName = landingZone.getStorageBucket().name();
+      HeadBucketRequest request = HeadBucketRequest.builder().bucket(bucketName).build();
+
+      // SdkClient classes should generally be used in try-with-resource blocks.
+      try (S3Client s3Client =
+          S3Client.builder().region(region).credentialsProvider(awsCredentialsProvider).build()) {
+        s3Client.headBucket(request);
+      } catch (Exception exception) {
+        fail(String.format("HeadBucket request failed: %s", exception), exception);
+      }
+
+      logger.info("Confirmed access to bucket {} in AWS Region {}", bucketName, region);
```

Comment:
logging is not necessary in test

Reply:
This is an existing test. It shows up as a new file only because the file name was changed. hence retained